### DISCUSSION TITLE
What is the purpose of this?

### DISCUSSION THREAD

**Discussion by sharpninja** [google/zx#517] [General]
I truly don't get it.  This is essentially PowerShell implemented in JavaScript.  I cannot imagine anyone who has actually used PowerShell looking at this and thinking it's an improvement in any way.  It does one thing for PowerShell, though, and that is to prove Microsoft got it right back around 2005 when PowerShell was released.

**Comment by antonmedv** [comment]
PowerShell is great. But have one disadvantage: I don’t know it. 

Same goes with Bash: I have to google how to return an array from a function every time. 

JS, on other hand, I remember in the middle of the night at 3a.m.  And if I need to do some lightweight scripting, it is perfectly. 

Unfortunately, nodejs api to call external programs is not great. Here zx comes and brings nicer api for calling bash from js. 

As JS is the most popular programming language, this specific use case (calling bash from js) is appealing to lots and lots of people. 

Hope it’s explains something)

PS Argument what I can learn PowerShell is not working, as it can be applied to anything:  but you can learn Babashka, Julia, Etc!

**Comment by luckman212** [comment]
Is there any reason to use this if you consider yourself a Bash "expert"? I was also surprised at close to 40K stars for a project that seems to be basically a JS wrapper around shell. I'm sure I'm missing something...

**Reply by antonmedv** [reply]
Q: how to return an array from a function in a bash?

I have to google it every time. But JS i know pretty well without googling. But nodejs lacks a nice interface to call external bins. Zx comes here.

**Reply by luckman212** [reply]
Ok, guess I'll start with [this search][1] and see if I can figure out by example when this should be used.

[1]: https://github.com/search?q=%22%23%21%2Fusr%2Fbin%2Fenv+zx%22+path%3A*.mjs&type=code

**Comment by BlackGlory** [comment]
Just imagine a programmer who takes no notes at all, relies on Google to find the answers to all his questions, and still chooses Google over taking notes the 100th time he encounters the same problem.

**Comment by kenkku** [comment]
I think the answer is clear: you are obviously not looking for an answer. There's plenty of libraries I don't like but I don't go around announcing it to random people on the internet. We. Don't. Care.
